vertical-pod-autoscaler/pkg/utils/test/test_vpa.go

```diff
@@ -248,7 +248,12 @@ func (b *verticalPodAutoscalerBuilder) Get() *vpa_types.VerticalPodAutoscaler {
 			Mode:             b.scalingMode[containerName],
 		})
 	}
-	recommendation = b.recommendation.WithContainer(b.containerNames[0]).Get()
+	// VPAs with a single container may still use the old/implicit way of adding recommendations
+	r := b.recommendation.WithContainer(b.containerNames[0]).Get()
```

Comment:
Why do we hardcode the container name index here and append to `recommendation.ContainerRecommendations` instead of doing this logic for all containers in the loop above?

Reply:
See comment above, probably the same issue/root cause?